test/unitary/generator/dependencies.cpp

```diff
@@ -0,0 +1,231 @@
+// Copyright 2020 Proyectos y Sistemas de Mantenimiento SL (eProsima).
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <gtest/gtest.h>
+#include <xtypes/idl/idl.hpp>
+#include <iostream>
+
+using namespace eprosima::xtypes;
+
+void generation_roundtrip_test(
+        const std::vector<std::pair<std::string, std::string>>& module_elements)
+{
+    Module root;
+    Module& mod_A = root.create_submodule("A");
+    Module& mod_B = root.create_submodule("B");
+    Module& mod_AB = mod_A.create_submodule("B");
+    Module& mod_AC = mod_A.create_submodule("C");
+    Module& mod_ABA = mod_AB.create_submodule("A");
+
+    const std::map<const std::string, Module&> modules = {
+        {"A",       mod_A  },
+        {"B",       mod_B  },
+        {"A::B",    mod_AB },
+        {"A::C",    mod_AC },
+        {"A::B::A", mod_ABA}
+    };
+
+    EnumerationType<uint32_t> my_enum("MyEnum");
+    my_enum.add_enumerator("A", 0);
+    my_enum.add_enumerator("B", 1);
+    my_enum.add_enumerator("C", 2);
+
+    DynamicData my_enum_data(my_enum);
+    my_enum_data = my_enum.value("B");
+
+    EnumerationType<uint32_t> my_union_enum("MyDiscriminatorEnum");
+    my_union_enum.add_enumerator("Disc_1", 1);
+    my_union_enum.add_enumerator("Disc_2", 2);
+    my_union_enum.add_enumerator("Disc_3", 5);
+    my_union_enum.add_enumerator("Disc_4", 7);
+
+    AliasType my_bool_alias(primitive_type<bool>(), "MyBoolAlias");
+    AliasType my_bool_alias_alias(my_bool_alias, "MyBoolAliasAlias");
+
+    StructType my_struct("MyStruct");
+    my_struct.add_member(Member("ms0", primitive_type<uint16_t>()));
+    my_struct.add_member("ms1", my_enum);
+
+    StructType my_child_struct("MyChildStruct", &my_struct);
+    my_child_struct.add_member("mcs0", StringType(20));
+
+    UnionType my_union("MyUnion", my_union_enum);
+    std::vector<std::string> labels = {"Disc_1", "Disc_2"};
+    my_union.add_case_member(labels, Member("my_union_alias", my_bool_alias_alias));
+    std::vector<uint32_t> test(my_union_enum.value("Disc_3"));
+    my_union.add_case_member(my_union_enum.value("Disc_3"), "my_union_float", primitive_type<float>());
+    my_union.add_case_member(my_union_enum.value("Disc_4"), "my_union_struct", my_struct, true);
```

Comment:
```suggestion
    my_union.add_case_member({my_union_enum.value("Disc_3")}, Member("my_union_float", primitive_type<float>()));
    my_union.add_case_member({my_union_enum.value("Disc_4")}, Member("my_union_struct", my_struct, true));
```

Reply:
For me, it is kind of weird to force the user use a vector to add a case member with just one label. What about
```
template<typename T>
UnionType& UnionType::add_case_member(const T& label, const Member& member)
```
instead of the one I introduced in this PR?